modules/terraform-aws-alternat/main.tf

```
@@ -280,6 +280,14 @@ resource "aws_security_group_rule" "nat_instance_ingress" {
   source_security_group_id = local.nat_instance_ingress_sgs[count.index]
 }
 
+resource "aws_security_group_rule" "nat_instance_ip_range_ingress" {
+  type                     = "ingress"
+  protocol                 = "-1"
+  from_port                = 0
+  to_port                  = 0
+  security_group_id        = aws_security_group.nat_instance.id
+  cidr_blocks = var.ingress_security_group_cidr_blocks
+}
```

Needs a guard to avoid creating this resource if no CIDR blocks are provided.

```suggestion
resource "aws_security_group_rule" "nat_instance_ip_range_ingress" {
  count = length(var.ingress_security_group_cidr_blocks) > 0 ? 1 : 0
  
  type                     = "ingress"
  protocol                 = "-1"
  from_port                = 0
  to_port                  = 0
  security_group_id        = aws_security_group.nat_instance.id
  cidr_blocks = var.ingress_security_group_cidr_blocks
}
```